packages/pds/src/db/db.ts

```diff
@@ -73,13 +76,15 @@ export class Database<Schema> {
   }
 
   async transaction<T>(fn: (db: Database<Schema>) => Promise<T>): Promise<T> {
-    return retry(() => this.transactionNoRetry(fn), {
-      retryable: (err) =>
-        typeof err?.['code'] === 'string' && RETRY_ERRORS.has(err['code']),
-      maxRetries: 5,
-      backoffMultiplier: 50,
-      backoffMax: 2000,
-    })
+    return retrySqlite(() => this.transactionNoRetry(fn))
+  }
+
+  async executeWithRetry<T>(query: { execute: () => Promise<T> }) {
```

Comment:
any reason for this syntax instead of a call back?

not really opposed, but it may be more inline with the rest of the db code as a call back 🤔 

if we wanted to get fancy, could even pass the underlying kysely instance into the callback and avoid the `db.db` 😛

Reply:
oh actually... since it doesn't include the `execute`, i can see that this pattern encourages you to not do more than one write at a time in the retry 🤔

Reply:
I know what you mean, went back and forth on it a little.  But yeah two things here, neither of which is crucial:
 - We only want to retry individual statements, and this allows us to enforce that.
 - Small thing, but we only need to compile the query one time no matter the number of retries. Also, when the statement is already in a transaction the no-op here is extremely cheap.

Reply:
nice i dig 👍